File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기-1.js
function solution(myString, pat) {
  // myString 의 부분문자열중 pat 으로 끝나는 가장 긴 문자열 리턴
  let temp = [];
  for (let i = 0; i < myString.length; i++) {
    let subStr = myString.slice(0, i + 1);
    if (subStr.endsWith(pat)) temp.push(subStr);
  }
  return temp.pop();
}

File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기-1.java
import java.util.ArrayList;

class Solution {
	public String solution( String myString, String pat ) {
		ArrayList< String > result = new ArrayList<>( );
		for ( int i = 0; i < myString.length( ); i++ ) {
			result.add( myString.substring( 0, i + 1 ) );
		}
		return result.get( result.size( ) - 1 );
	}
}